toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```
@@ -0,0 +1,222 @@
+		if strings.HasPrefix(trimmedLine, "linux ") {
+			// Append new arguments to the line that starts with "linux"
+			line += " " + newArgs
+		}
+		if strings.HasPrefix(trimmedLine, "set rootdevice=PARTUUID=") {
+			// Replace the root device line with the new root device. TODO: add supported type 'user'
+			line = "set rootdevice=/dev/mapper/root"
+		}
+		updatedLines = append(updatedLines, line)
+	}
+
+	// Write the updated content back to grub.cfg
+	err = ioutil.WriteFile(grubConfigPath, []byte(strings.Join(updatedLines, "\n")), 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write updated grub config: %v", err)
+	}
+
+	return nil
+}
+
+// findFreeNBDDevice finds the first available NBD device.
+func findFreeNBDDevice() (string, error) {
+	files, err := filepath.Glob("/sys/class/block/nbd*")
+	if err != nil {
+		return "", err
+	}
+
+	for _, file := range files {
+		size, err := ioutil.ReadFile(filepath.Join(file, "size"))
+		if err != nil {
+			continue
+		}
+
+		if strings.TrimSpace(string(size)) == "0" {
+			return "/dev/" + filepath.Base(file), nil
+		}
+	}
+
+	return "", fmt.Errorf("no free nbd devices available")
+}
+
+// convertToDevicePath takes an NBD device path and a system config device path,
+// and converts it to the corresponding NBD partition path.
+func convertToNbdDevicePath(nbdDevice, systemDevice string) (string, error) {
+	partitionRegex := regexp.MustCompile(`[0-9]+$`)
+	partitionNumber := partitionRegex.FindString(systemDevice)
+
+	if partitionNumber == "" {
+		return "", fmt.Errorf("failed to extract partition number from %s", systemDevice)
+	}
+
+	return fmt.Sprintf("%sp%s", nbdDevice, partitionNumber), nil
+}
+
+// findDeviceByUUIDOrLabel attempts to resolve a PARTUUID, PARTLABEL, UUID, or LABEL to a device file.
+func findDeviceByUUIDOrLabel(uuidOrLabel string) (string, error) {
+	// Check if the input is already a device path
+	if strings.HasPrefix(uuidOrLabel, "/dev/") {
+		return uuidOrLabel, nil
```

Should throw an error if this is the case.

---

It would be a logical error if this function is called with an already-resolved /dev path.